Papr/Scenes/Search/SearchPhotos/Cell/SearchPhotosCellModel.swift

```diff
@@ -23,11 +26,14 @@ protocol SearchPhotosCellModelType {
 class SearchPhotosCellModel: SearchPhotosCellModelType,
                             SearchPhotosCellModelInput,
                             SearchPhotosCellModelOutput {
-
+    
     // MARK: Inputs & Outputs
     var inputs: SearchPhotosCellModelInput { return self }
     var outputs: SearchPhotosCellModelOutput { return self }
 
+    // MARK: Inputs
+    var photoSize: Variable<CGSize> = Variable(.zero)
```

Comment:
Remove this.

Reply:
@jdisho what do you mean remove it, to remove the initialisation or
I have no idea how otherwise to send events with size to update the CollectionView
(so I made public property)

Reply:
`photoSize` should be an output (Observable<(width: Double, height: Double>), not an input.

Reply:
That's how the viewModel should look like and btw the viewModels should not have anything related to UIKit.

```swift 
import Foundation
import RxSwift

protocol SearchPhotosCellModelInput {
    func updateSize(width: Double, height: Double)
}

protocol SearchPhotosCellModelOutput {
    var smallPhotoURL: Observable<String> { get }
    var regularPhotoURL: Observable<String> { get }
    var photoSize: Observable<(width: Double, height: Double)> { get }
}

protocol SearchPhotosCellModelType {
    var inputs: SearchPhotosCellModelInput { get }
    var outputs: SearchPhotosCellModelOutput { get }
}

class SearchPhotosCellModel: SearchPhotosCellModelType,
                            SearchPhotosCellModelInput,
                            SearchPhotosCellModelOutput {
    
    // MARK: Inputs & Outputs
    var inputs: SearchPhotosCellModelInput { return self }
    var outputs: SearchPhotosCellModelOutput { return self }

    // MARK: Inputs
    func updateSize(width: Double, height: Double) {
        photoSizeProperty.onNext((width: width, height: height))
    }

    // MARK: Outputs
    let smallPhotoURL: Observable<String>
    let regularPhotoURL: Observable<String>
    let photoSize: Observable<(width: Double, height: Double)>

    // MARK: Privates
    private let photoSizeProperty = BehaviorSubject<(width: Double, height: Double)>(value: (width: 0, height: 0))

    // MARK: Init
    init(photo: Photo) {
        let photoStream = Observable.just(photo)

        smallPhotoURL = photoStream
            .map { $0.urls?.small }
            .unwrap()

        regularPhotoURL = photoStream
            .map { $0.urls?.regular }
            .unwrap()

        photoSize = photoSizeProperty.asObservable()
    }
}

```

Reply:
😃 thanks for detailed hint, just made everything really clear 
just one more question you want to get rid of all UIKit in models, so you can test them right?